### DISCUSSION TITLE
Help with conditionally rendering replicator

### DISCUSSION THREAD

**Discussion by suarez2145** [statamic/cms#4937] [Q&A] [answered]
Hi Im pretty new to Statamic and i am stuck on something. My issue is, I currently have a {{REPLICATOR}} wrapped in a div in my antlers.html file and i want to know if i can check if the {{REPLICATOR}} column fields have data from OUTSIDE the {{ REPLICATOR }} antlers. I want to do this because the {{ REPLICATOR}} is wrapped in a div and when there is no data in the column fields the div shows up and takes up space on my page and it interferes with my HERO image. 

EXAMPLE: 
   ** if there is no data in the REPLICATOR can i remove the  outside  -- div -- ??
 -- div --
   {{ REPLICATOR }}
       {{ REPLICATOR COLUMN CONTENT }}
   {{ / REPLICATOR}}
-- /div --

**Comment by JohnathonKoster** [comment, marked as answer]
There are two approaches to take here. The easiest method is to simply check the variable with an `if` statement. `null` or empty arrays will be treated as `false`, and will cause the code inside the if statement to be ignored:

```antlers
{{ if replicator }}
    <div>
        {{ replicator }}
            <!-- Normal Replicator Template Code Here -->
        {{ /replicator }}
    </div>
{{ /if }}
```

The second is to use the [as](https://statamic.dev/modifiers/as) modifier to move the replicator's sets to a different variable name, allowing us to move the div inside the `{{ replicator }}`. We can then wrap our div inside an if statement (if sets is `null` or empty, it will evaluate to false):

```antlers
{{ replicator as="sets" }}
    {{ if sets }}
        <div>
            {{ sets }}
            <!-- Normal Replicator Template Code Here -->
            {{ /sets }}
        </div>
    {{ /if }}
{{ /replicator }}
```

**Reply by suarez2145** [reply]
This solved my issue. Thanks
